test/java/src/com/ibm/streamsx/rest/test/ParallelOperatorsTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2017
+ */
+
+package com.ibm.streamsx.rest.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.ibm.streamsx.rest.Operator;
+import com.ibm.streamsx.topology.TStream;
+import com.ibm.streamsx.topology.Topology;
+import com.ibm.streamsx.topology.context.StreamsContext;
+import com.ibm.streamsx.topology.context.StreamsContextFactory;
+
+public class ParallelOperatorsTest {
+
+    StreamsConnectionTest helper;
+
+    @After
+    public void removeJob() throws Exception {
+        helper.removeJob();
+    }
+
+    @Before
+    public void setupJob() throws Exception {
+        if (null == helper) {
+            helper = new StreamsConnectionTest();
+        }
+        helper.setupInstance();
+        if (helper.jobId == null) {
+            // avoid clashes with sub-class tests
+            Topology topology = new Topology(getClass().getSimpleName(), 
+                    "JobForRESTApiTest");
+
+            TStream<Integer> source = topology.periodicSource(() -> (int) (Math.random() * 5000 + 1), 200, TimeUnit.MILLISECONDS);
+            source.invocationName("IntegerPeriodicMultiSource");
+            TStream<Integer> parallelSource = source.parallel(3);
+            TStream<Integer> sourceDouble = parallelSource.map(StreamsConnectionTest.doubleNumber());
+            sourceDouble.invocationName("Outer");
+            TStream<Integer> encore = sourceDouble.parallel(2);
```

Comment:
Interesting to see that this worked, since technically we don't officially support nested parallelism in the API. We don't preclude it, but it hasn't been tested.